iree/tools/iree-e2e-matmul-test.c

```diff
@@ -515,29 +515,20 @@ static iree_status_t allocate_buffer_like(iree_hal_allocator_t* hal_allocator,
       iree_hal_buffer_view_element_type(src),
       iree_hal_buffer_view_encoding_type(src),
       (iree_hal_buffer_params_t){
-          .type = IREE_HAL_MEMORY_TYPE_HOST_LOCAL |
-                  IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE,
-          .usage = IREE_HAL_BUFFER_USAGE_DISPATCH |
-                   IREE_HAL_BUFFER_USAGE_TRANSFER |
-                   IREE_HAL_BUFFER_USAGE_MAPPING,
+          .type = IREE_HAL_MEMORY_TYPE_HOST_LOCAL,
+          .usage =
+              IREE_HAL_BUFFER_USAGE_TRANSFER | IREE_HAL_BUFFER_USAGE_MAPPING,
       },
       iree_const_byte_span_empty(), dst);
 }
 
-// Performs a deep copy of |src| into |dst|. Takes care of allocating |dst|.
-static iree_status_t copy_buffer(iree_hal_allocator_t* hal_allocator,
+// Performs a deep copy of |src| into a host local |dst| buffer. Takes care of
+// allocating |dst|.
+static iree_status_t copy_buffer(iree_hal_device_t* device,
```

Comment:
naming nit: rename to `clone_buffer` so it's clearer this is allocating a new buffer